pkg/minikube/node/cache.go

```diff
@@ -99,14 +100,20 @@ func doCacheBinaries(k8sVersion string) error {
 }
 
 // BeginDownloadKicArtifacts downloads the kic image + preload tarball, returns true if preload is available
-func beginDownloadKicArtifacts(g *errgroup.Group, driver string, cRuntime string, baseImage string) {
-	glog.Infof("Beginning downloading kic artifacts for %s with %s", driver, cRuntime)
-	if driver == "docker" {
-		if !image.ExistsImageInDaemon(baseImage) {
+func beginDownloadKicArtifacts(g *errgroup.Group, cc *config.ClusterConfig) {
+	glog.Infof("Beginning downloading kic artifacts for %s with %s", cc.Driver, cc.KubernetesConfig.ContainerRuntime)
+	if cc.Driver == "docker" {
+		if !image.ExistsImageInDaemon(cc.KicBaseImage) {
 			out.T(out.Pulling, "Pulling base image ...")
 			g.Go(func() error {
-				glog.Infof("Downloading %s to local daemon", baseImage)
-				return image.WriteImageToDaemon(baseImage)
+				glog.Infof("Downloading %s to local daemon", cc.KicBaseImage)
```

Comment:
Please add TODO to respect the `--image-repository` flag.

Reply:
done